site/static/api/device-registry-v1.yaml

```diff
@@ -0,0 +1,724 @@
+         description: |
+             Deletes all credentials from a device registration. The device is
+             still registered, and a following call to *get* credentials must
+             not return `404`, as the device is still registered. Instead an
+             empty result must be returned.
+         operationId: deleteAllCredentials
+         parameters:
+            - $ref: '#/components/parameters/resourceVersion'
+         responses:
+            204:
+               $ref: '#/components/responses/Delete'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            404:
+               $ref: '#/components/responses/NotFound'
+            412:
+               $ref: '#/components/responses/ResourceVersionMismatch'
+
+components:
+
+   schemas:
+
+# Common schema
+
+      Error:
+         type: object
+         required:
+            - error
+         properties:
+            "error":
+               type: string
+               description: A human readable error message of what went wrong.
+
+      Extensions:
+         type: object
+         additionalProperties: true
+         description: Allows arbitrary properties as extension to the ones
+                      specified by the Hono API.
+
+# Tenant schema
+
+      Tenant:
+         type: object
+         additionalProperties: false
+         properties:
+            "enabled":
+               type: boolean
+               default: true
+            "ext":
+               $ref: '#/components/schemas/Extensions'
+            "adapters":
+               type: array
+               description: Only a single entry per type is allowed. If multiple entries for the same type are present it is handled as an error.
+               items:
+                  $ref: '#/components/schemas/Adapter'
+            "limits":
+               $ref: '#/components/schemas/ResourceLimit'
+            "trusted-ca":
+               $ref: '#/components/schemas/TrustedCA'
+
+      TrustedCA:
+         type: object
+         additionalProperties: false
+         required:
+            - "subject-dn"
+         properties:
+            "subject-dn":
+               type: string
+            "public-key":
+               type: string
+               format: byte
+            "cert":
+               type: string
+               format: byte
+
+      Adapter:
+         type: object
+         additionalProperties: true
+         required:
+            - type
+         properties:
+            "type":
+               type: string
+            "enabled":
+               type: boolean
+               default: false
+            "device-authentication-required":
+               type: boolean
+               default: true
+            "ext":
+               $ref: '#/components/schemas/Extensions'
+
+      ResourceLimit:
+         type: object
+         additionalProperties: false
+         properties:
```

Comment:
IMHO we should allow registration of additional/custom limits here via the *ext* mechanism

Reply:
Done.